mlrun/api/utils/clients/iguazio.py

```diff
@@ -505,11 +503,34 @@ def _generate_auth_info_from_session_verification_response(
         if x_unix_uid and x_unix_uid.lower() != "unknown":
             user_unix_id = int(x_unix_uid)
 
+        # get user and group ids from response body
+        response_body = response.json()
+        context_auth = (
+            response_body.get("data", {})
+            .get("attributes", {})
+            .get("context", {})
+            .get("authentication", {})
+        )
+        user_id = context_auth.get("user_id", None)
+        if user_id is None:
+            user_id = response.headers.get("x-user-id")
+        gids = context_auth.get("group_ids", None)
+        group_ids = []
+        if gids is None:
+            gids = response.headers.get("x-user-group-ids")
+            # "x-user-group-ids" header is a comma separated list of group ids
+            if gids:
+                group_ids = gids.split(",")
+        else:
+            for gid in gids:
+                # some gids can be a comma separated list of group ids
+                group_ids.extend(gid.split(","))
+
```

Comment:
Tried to make this section more concise and cleaner, lmk what yo think
```suggestion
        gids = context_auth.get("group_ids", [])
        
        # fallback to header if no GIDs in the body
        if not gids:
            gids = response.headers.get("x-user-group-ids")
            # "x-user-group-ids" header is a comma separated list of group ids
            if gids:
                gids = gids.split(",")
        
        # some gids can be a comma separated list of group ids (if taken from the headers, the next split will have no effect)
        group_ids = [gid.split(",") for gid in gids]
```